src/AnnotationThread.js

```diff
@@ -115,66 +122,105 @@ class AnnotationThread extends EventEmitter {
         this.state = STATES.inactive;
     }
 
-    /**
-     * Whether or not thread dialog is visible
-     *
-     * @return {void}
-     */
-    isDialogVisible() {
-        return !!(this.dialog && this.dialog.element && !this.dialog.element.classList.contains(CLASS_HIDDEN));
-    }
+    position = () => {};
 
     /**
      * Shows the appropriate annotation dialog for this thread.
      *
+     * @param {Event} event - Mouse event
      * @return {void}
      */
-    showDialog() {
-        // Prevents the annotations dialog from being set up on every call
-        if (!this.dialog.element) {
-            this.dialog.setup(this.annotations, this.element);
+    renderAnnotationPopover(event = null) {
+        if (event) {
+            event.stopPropagation();
+            event.preventDefault();
+        }
+
+        const firstAnnotation = this.annotations[0];
+        const commentAnnotations = this.annotations.filter(
+            (annotation) => annotation.message && annotation.message !== ''
+        );
+        const comments = commentAnnotations.map((comment) => {
+            const { id, message, permissions, createdBy, createdAt } = comment;
+            return {
+                id,
+                message,
+                permissions,
+                createdBy,
+                createdAt
+            };
+        });
+
+        const pageEl = this.annotatedElement.querySelector(`[data-page-number="${this.location.page}"]`);
+        const annotationDialogLayer = pageEl.querySelector('.ba-dialog-layer');
+
+        const isPending = !firstAnnotation;
+        if (isPending) {
+            this.state = STATES.pending;
+        } else {
+            this.state = STATES.active;
         }
 
-        this.dialog.show(this.annotations);
+        const hasComments = comments.length > 0;
+        this.popoverComponent = render(
+            <AnnotationPopover
+                id={hasComments || isPending ? this.threadID : firstAnnotation.id}
+                type={this.type}
+                createdAt={get(firstAnnotation, 'createdAt', null)}
+                createdBy={get(firstAnnotation, 'createdBy', null)}
+                modifiedBy={get(firstAnnotation, 'modifiedBy', null)}
+                canAnnotate={this.permissions.canAnnotate}
```

Comment:
What's the difference between canAnnotate and canComment here? If the annotation thread already exists don't we just need can comment and can delete?

Reply:
canComment determines whether or not the highlight dialog can add comments to it or not. 
canAnnotate determines whether or not you can create new annotations